CakePHP/Sniffs/Commenting/TypeHintSniff.php

```diff
@@ -86,32 +117,35 @@ public function process(File $phpcsFile, $stackPtr)
         foreach ($tokens[$stackPtr]['comment_tags'] as $tag) {
             if (
                 $tokens[$tag + 2]['code'] !== T_DOC_COMMENT_STRING ||
-                !in_array($tokens[$tag]['content'], self::$typeHintTags, true)
+                !in_array($tokens[$tag]['content'], static::$typeHintTags, true)
             ) {
                 continue;
             }
 
             $tagComment = $phpcsFile->fixer->getTokenContent($tag + 2);
-            $valueNode = self::getValueNode($tokens[$tag]['content'], $tagComment);
+            $valueNode = static::getValueNode($tokens[$tag]['content'], $tagComment);
             if ($valueNode instanceof InvalidTagValueNode) {
                 continue;
             }
 
+            $hasUnion = false;
+            /** @var \PHPStan\PhpDocParser\Ast\PhpDoc\PropertyTagValueNode|\PHPStan\PhpDocParser\Ast\PhpDoc\VarTagValueNode|\PHPStan\PhpDocParser\Ast\PhpDoc\ParamTagValueNode|\PHPStan\PhpDocParser\Ast\PhpDoc\ReturnTagValueNode $valueNode */
             if ($valueNode->type instanceof UnionTypeNode) {
                 $types = $valueNode->type->types;
+                $hasUnion = true;
             } elseif ($valueNode->type instanceof ArrayTypeNode) {
                 $types = [$valueNode->type];
             } else {
                 continue;
             }
 
             $originalTypeHint = $this->renderUnionTypes($types);
-            $sortedTypeHint = $this->getSortedTypeHint($types);
+            $sortedTypeHint = $this->getSortedTypeHint($types, $hasUnion);
```

Comment:
Maybe should split up the array conversion and sorting so this can be called just for union types.